sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/ExpressionParserSuite.scala

```diff
@@ -632,7 +632,7 @@ class ExpressionParserSuite extends AnalysisTest {
 
     // Non Existing unit
     intercept("interval 10 nanoseconds",
-      "no viable alternative at input 'interval 10 nanoseconds'")
+      "missing {'DAY', 'DAYS', 'HOUR'", "'YEAR', 'YEARS'} at 'nanoseconds'")
```

Comment:
The new message is better. Users can turn it into a valid SQL like `interval 10 day nanoseconds`, which creates a interval literal and give an alias.